### ISSUE TITLE
Alternative to QR-code to link devices without camera

### ISSUE THREAD

**Issue by valldrac** [mollyim/mollyim-android#46]

**Comment by haarp** [comment]
Seconded. I am dealing with a device with a broken camera focus, and thus cannot link another device. Infuriating decision on Sginal's part. See https://github.com/signalapp/Signal-Android/issues/5153

Someone already started work on a workaround, see https://github.com/signalapp/Signal-Android/pull/5538

**Comment by valldrac** [comment]
I see. I will try to work on this for the next release.

**Comment by haarp** [comment]
Got to try this the other day in the new release. Scanned the PC app's QR code with a different device and entered the resulting string manually in Molly. Success!

Thanks a lot!

**Comment by straussmaus** [comment]
Do you mean you're cross-communicating between Signal and mollyim?

**Comment by haarp** [comment]
> Is the thing you got working good for this situation?

Yes, that's exactly the use case. Scan the QR code in signal-desktop with anything that works, which will give you a string. Enter this string manually into Molly and voila.

**Comment by hsnyder** [comment]
Is this functionality still exposed, or has it been overwritten since it was released? I installed the latest Molly from F-Droid, and I couldn't see any way to do this. Sorry for re-opening.

**Comment by 9iva** [comment]
Harris Snyder I hope this to be ur true name. I get attached to lies
finding my truth I. Their blood. I am not light I am alive can you tell me
what your name is

**Comment by valldrac** [comment]
> I installed the latest Molly from F-Droid, and I couldn't see any way to do this.

Go to Linked Devices in settings, tap on the add icon, and there you have the option "Configure without scanning".

**Comment by code-mart** [comment]
Hi @valldrac, I have the same experience as @hsnyder, as there is no "Configure without scanning" option that appears. Only a black square and message that says "Scan the QR code displayed on the device to link." Version is 5.34.9-1-FOSS. Thanks for any insight.

**Comment by code-mart** [comment]
Just tried again after re-installing with non-FOSS release, v5.34.9-2.apk. 

Same result. Cannot see any "Configure without scanning" option.

**Comment by valldrac** [comment]
This is the screen to link devices. And at the bottom is the "Link without scanning" button. Could it be that the screen of your device is not large enough and the button is hidden?

![image](https://user-images.githubusercontent.com/10455969/162075452-785aa1a0-e9b3-4c90-bceb-db6787f3c810.png)

**Comment by code-mart** [comment]
@valldrac Thank you so much! Yes, that was indeed the problem. Going into (android settings-->Display-->Advanced-->Display Size) and reducing the size allowed me to see the "Link without scanning" button.

This experience has made me appreciate Molly hugely. Seeing how the Signal team keeps such a helpful and easily-implemented feature locked down is a perfect example of why a community-maintained fork is so necessary. Thank you for all your efforts!

**Comment by code-mart** [comment]
Hi @valldrac, I am having trouble pasting into the "Link this device?" text box, so I am unable link by copy and paste. Maybe this would be an easy fix for MollyIM? Or perhaps there is a workaround? Thanks!

**Comment by code-mart** [comment]
Nevermind, I was running this in Waydroid, and I did not notice that the clipboard was not working at all. I fixed by installing pyclips and wl-clipboard, as explained in the issues below. Sorry for the trouble and thanks for the response!

https://github.com/waydroid/waydroid/issues/276
https://github.com/waydroid/waydroid/issues/309

**Comment by pets-er** [comment]
Hi there!

Maybe I just can't see where the "Link without scanning" button is (I did reduce display size just to be sure).
However, it appears that the linking dialogue has changed significantly upstream.
As such I think this issue should be re-opened - or at least investigated with regards to whether other people do not see the button either.
Also: the button should probably show up without granting camera permission.

Thank you for your work!
